Updated the Android build images.

FenrirWillow opened the pull request:
The build images are now separated per SDK version and also contain 'slim' versions without the NDK inside them. This significantly improves their size and build times. Also all tools are not at their latest versions including the way we manage the SDKs and packages within the Android ecosystem

FenrirWillow committed:
Updated the Android build images.

The build images are now separated per SDK version and also contain 'slim' versions without the NDK inside them. This significantly improves their size and build times. Also all tools are not at their latest versions including the way we manage the SDKs and packages within the Android ecosystem

FenrirWillow commented:
I don't know what the best way to handle the latest tag is. I think we should just not have it, since it's not really something incremental. Would appreciate some input on it.

dannytip commented:
You should probably tag it as per: https://github.com/phusion/passenger-docker

Actually have separate repo's for the different versions.  Since they are public repos it shouldn't be a problem.

I.e.

phusion/passenger-ruby20:<VERSION>
phusion/passenger-ruby21:<VERSION>
phusion/passenger-ruby22:<VERSION>

dannytip commented on android-build/sdk-23/Dockerfile:
```diff
@@ -1,95 +1,96 @@
+
+# --- Run system tool updates ---
+RUN apt-get update
+RUN apt-get -y install unzip
```
Any reason to not just bundle all these into 1 line?

FenrirWillow commented:
About the tagging and the different versions - I prefer keeping all versions in the same repository, this way we can build them with a simple bash script.

The script also does the tagging for us. Currently it will tag the images like this: `sdk-<version>` or `sdk-<version>-ndk` for the images that have the ndk inside them.

dannytip commented:
I'm sure you could still build them with a simple bash script if they were in different repos (docker repo's that is, the source could all stay in one).

@pmead what do you think?

FenrirWillow committed:
Fixed user permission issues by adding a new user which is actually executing builds.

The issue was stemming from the fact that the volume would contain files owned by the root user (UID 1) and the host OS wouldn't always be able to clean those up. This is no longer a problem if we run the build as another temporary user.
